spanner_orm/condition.py

```diff
@@ -225,13 +225,67 @@ def _validate(self, model):
     del model
 
 
+class OrCondition(Condition):
+  """Used to join multiple conditions with an OR in a Spanner query."""
+
+  def __init__(self, *condition_lists):
+    super().__init__()
+    if len(condition_lists) < 2:
+      raise error.SpannerError(
+          'OrCondition requires at least two lists of conditions')
+    self.condition_lists = condition_lists
+    self.all_conditions = []
+    for conditions in condition_lists:
+      self.all_conditions.extend(conditions)
+
+  def bind(self, model):
+    super().bind(model)
+    for condition in self.all_conditions:
+      condition.bind(model)
+
+  def _params(self):
+    result = {}
+    for condition in self.all_conditions:
+      condition.suffix = str(int(self.suffix or 0) + len(result))
+      result.update(condition.params())
+    return result
+
+  def _sql(self):
+    segments = []
+    params = 0
+    # Set proper suffix for each condition first
+    for condition in self.all_conditions:
+      condition.suffix = str(params + int(self.suffix or 0))
+      params += len(condition.params())
+
+    for conditions in self.condition_lists:
+      new_segment = ' AND '.join([condition.sql() for condition in conditions])
+      segments.append('({new_segment})'.format(new_segment=new_segment))
+    return '({segments})'.format(segments=' OR '.join(segments))
+
+  @staticmethod
+  def segment():
+    return Segment.WHERE
+
+  def _types(self):
+    result = {}
+    for condition in self.all_conditions:
+      condition.suffix = str(int(self.suffix or 0) + len(result))
+      result.update(condition.types())
+    return result
+
+  def _validate(self, model):
+    # condition is valid if all child conditions are valid
```

Comment:
Why aren't we explicitly validating all child conditions?

Reply:
validate is called as part of bind() (because validating requires a model to be associated with the condition, which happens at bind time), so they all are validated... just not here